src/EuroDNS/Provider.php

```diff
@@ -0,0 +1,428 @@
+
+        // Call the API to get poll messages based on parameters
+        $poll = $this->api()->getPollMessages(intval($params->limit), $since);
+
+        // Check if there is an error in the poll response
+        if(isset($poll['error'])){
+            // Throw an exception with the error message
+            throw $this->errorResult(sprintf((string)$poll['msg']));
+        }
+
+        // Create a PollResult object from the poll response
+        return PollResult::create($poll);
+
+    }
+
+    /**
+     * Check the availability of multiple domains.
+     */
+    public function domainAvailabilityCheck(DacParams $params): DacResult
+    {
+        // Normalize second-level domain (SLD)
+        $sld = Utils::normalizeSld($params->sld);
+
+        // Create an array of full domain names by combining SLD with each TLD
+        $domains = array_map(
+            fn ($tld) => $sld . "." . Utils::normalizeTld($tld),
+            $params->tlds
+        );
+
+        // Call the API to check the availability of the specified domains
+        $domainsResponse = $this->api()->checkDomains($domains);
+
+        // Check if there is an error in the domains response
+        if(isset($domainsResponse['error'])){
+              // Throw an exception with the error message
+            throw $this->errorResult(sprintf((string)$domainsResponse['msg']));
+        }
+
+        // Create a DacResult object with the domain availability information
+        return DacResult::create([
+            'domains' => $domainsResponse,
+        ]);
+
+    }
+
+    /**
+     * Function to connect to API class file in Helper
+     */
+
+    protected function api(): EuroDNSApi
+    {
+        if (isset($this->api)) {
+            return $this->api;
+        }
+        return $this->api = new EuroDNSApi($this->configuration);
+    }
+
+
+    /**
+     * @inheritDoc
+     */
+    public function register(RegisterDomainParams $params): DomainResult
+    {
+        // Construct the full domain name from the second-level domain (SLD) and top-level domain (TLD)
+        $domainName = Utils::getDomain($params->sld, $params->tld);
+
+        // Validate and check registration parameters
+        $this->checkRegisterParams($params);
+
+        // Call the API to register the domain
+        $registerDomain = $this->api()->register($params);
+
+        // Check if there is no error during domain registration
+        if(!$registerDomain['error']){
+            //sometime   take  time to get domain info of newly added domain .So, we put a delay in the process
+            sleep(3);
+
+            // Retrieve and return domain information after successful registration
+           return $this->_getInfo($domainName, sprintf('Domain %s was registered successfully!', $domainName));
+
+        }else{
+            // Throw an exception with the error message if domain registration fails
+            throw $this->errorResult(sprintf($registerDomain['msg']));
```

Comment:
In general it may be helpful to pass additional error data to `$this->errorResult()` than just the error message, to assist staff with debugging e.g.,

```suggestion
            throw $this->errorResult((string)$registerDomain['msg'], ['response' => $registerDomain]);
```

Reply:
Updated for all error responses